packages/custom_lint/lib/src/workspace.dart

```diff
@@ -181,8 +181,15 @@ class CustomLintWorkspace {
 
     final contextRootsWithCustomLint = await Future.wait(
       allContextRoots.map((contextRoot) async {
-        final pubspecFile = Directory(contextRoot.root.path).pubspec;
-        if (!pubspecFile.existsSync()) {
+        final projectDir = tryFindProjectDirectory(
+          Directory(contextRoot.root.path),
+        );
+        if (projectDir == null) {
+          return null;
+        }
```

Comment:
Use a single line in those situations

```suggestion
        if (projectDir == null) return null;
```